portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -15,18 +15,74 @@
 package com.google.sps.servlets;
 
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+
+
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
-public class DataServlet extends HttpServlet {
+public class DataServlet extends HttpServlet {    
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        // Get all the comments from the database  
+        Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+
+        List<String> comments = new ArrayList<String>();
+        for (Entity entity : results.asIterable()) {
+            String comment = (String) entity.getProperty("comment");
+            comments.add(comment);
+        }
+
+        response.setContentType("application/json");
+        response.getWriter().println(toJSONString(comments));
+    }
+
+    /*
+    * Helper function to store comments in JSON format
+    */
+    private String toJSONString(List<String> array) {
+            String json = "{ \"comments\" :" + "\""+ array.toString() + "\""+ "}";
```

Comment:
Can any string be in the json array?
Can you think of some input that could break this?

Reply:
I should do a null check first. 
Can make it more general by passing in the name of the array to the function as well

Reply:
What if one of the attempted values would be any one of these:

- can't
- Loved "Citizen Kane"
- http://adgfzs
- yes\no

-

Reply:
Using Gson to parse the input now and all of these input strings are now valid